checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -4,88 +4,366 @@
-    private final Class<? extends Annotation> checkerLockHeldClass = org.checkerframework.checker.lock.qual.LockHeld.class;
+    private final Class<? extends Annotation> checkerGuardedByClass = GuardedBy.class;
+    private final Class<? extends Annotation> checkerGuardSatisfiedClass = GuardSatisfied.class;
 
-    // Note that Javax and JCIP @GuardedBy is used on both methods and objects. For methods they are
-    // equivalent to the Checker Framework @Holding annotation.
-    private final Class<? extends Annotation> javaxGuardedByClass = javax.annotation.concurrent.GuardedBy.class;
-    private final Class<? extends Annotation> jcipGuardedByClass = net.jcip.annotations.GuardedBy.class;
+    /** Annotation constants */
+    protected final AnnotationMirror GUARDEDBY, GUARDEDBYUNKNOWN, GUARDSATISFIED, GUARDEDBYBOTTOM;
 
     public LockVisitor(BaseTypeChecker checker) {
         super(checker);
 
+        GUARDEDBYUNKNOWN = AnnotationUtils.fromClass(elements, GuardedByUnknown.class);
+        GUARDEDBY = AnnotationUtils.fromClass(elements, GuardedBy.class);
+        GUARDSATISFIED = AnnotationUtils.fromClass(elements, GuardSatisfied.class);
+        GUARDEDBYBOTTOM = AnnotationUtils.fromClass(elements, GuardedByBottom.class);
+
         checkForAnnotatedJdk();
     }
 
     @Override
+    public Void visitVariable(VariableTree node, Void p) { // visit a variable declaration
+        // A user may not annotate a primitive type, a boxed primitive type or a String
+        // with any qualifier from the @GuardedBy hierarchy.
+
+        // TODO: Note that there is currently no way to reliably retrieve user-written
+        // qualifiers (i.e. there is no way to know whether the qualifier was defaulted by
+        // the Checker Framework) without using the node.toString() hack below.
+
+        TypeMirror nodeType = atypeFactory.getAnnotatedType(node).getUnderlyingType();
+        if ((TypesUtils.isBoxedPrimitive(nodeType) ||
+             TypesUtils.isPrimitive(nodeType) ||
+             TypesUtils.isString(nodeType)) &&
+            (node.toString().contains("GuardSatisfied") ||
+             node.toString().contains("GuardedBy"))){ // HACK!!! TODO: Fix once there is a way to reliably retrieve user-written qualifiers.
+            checker.report(Result.failure("primitive.type.guardedby"), node);
+        }
+        return super.visitVariable(node, p);
+    }
+
+    @Override
     public LockAnnotatedTypeFactory createTypeFactory() {
-        // We need to directly access useFlow from the checker, because this method gets called
-        // by the superclass constructor and a field in this class would not be initialized
-        // yet. Oh the pain.
-        return new LockAnnotatedTypeFactory(checker, true);
+        return new LockAnnotatedTypeFactory(checker);
+    }
+
+    // Issue an error if a method (explicitly or implicitly) annotated with @MayReleaseLocks has a formal parameter
```

Comment:
Make Javadoc.  There are other method comments in the file that need to be made into Javadoc too.